frontends/p4/fromv1.0/converters.h

```diff
@@ -681,10 +684,10 @@ class DetectDuplicates: public Inspector {
                     auto e2 = n->second;
                     if (e1->node_type_name() == e2->node_type_name()) {
                         if (e1->srcInfo.getStart().isValid())
-                            ::error("%1%: same name as %2%", e1, e2);
+                            ::error(ErrorType::ERR_DUPLICATE, "%1%: same name as %2%", e1, e2);
```

Comment:
should remove the initial `%1%` in the error message as will be duplicated?  Or is this error type different?

Reply:
I find this a bit confusing myself; some error messages prepend a prefix that contains an argument, and some do not. This is in `lib/error_catalog.cpp`. This actually makes porting the error messages quite painful.

Reply:
Feel free to change that and have the full message in the invocation. I added the prefix based on the suggestion that it will be more conducive to write consistent error messages. Not sure whether that is paying off.

Reply:
Sorry, I am late to the party, but when I was debugging `::error` in my PR yesterday, I understood the scoop.   If you see this message in my code:

https://github.com/p4lang/p4c/pull/2125/files#diff-e9f74513df031210ac7355c6025666a1R60

The args starts from `%2%` because `%1%` is used to print IR:Node which is IR::Range and the line number and filename of P4 program where the error occurred.   The output of the `::error` message is shown below.

https://github.com/p4lang/p4c/pull/2125/files#diff-a38ccf8a1b804434bb703693f1c5e66aR1